MapboxNavigation/RouteTableViewController.swift

```diff
@@ -46,6 +47,40 @@ class RouteTableViewController: UIViewController {
             headerView.timeRemaining.text = String.localizedStringWithFormat(NSLocalizedString("LESS_THAN", bundle: .mapboxNavigation, value: "<%@", comment: "Format string for less than; 1 = duration remaining"), dateComponentsFormatter.string(from: 61)!)
         } else {
             headerView.timeRemaining.text = dateComponentsFormatter.string(from: routeProgress.durationRemaining)
+            
+            guard let coordinates = routeProgress.route.coordinates else { return }
+            
+            // To keep this cheap, use the fraction traveled along the current route.
+            // From this, we can estimate what coordinates in the route remain.
+            // This color change does not need to be 100% accurate, 
+            // just a rough estimation of remaining route congestion.
+            let estimatedCoordinatesRemaining = Int(floor(Double(coordinates.count) * routeProgress.fractionTraveled))
```

Comment:
Hard to say. The number of coordinates in a given step has no relation to its distance. For example, a short, windy stretch of road may have many more coordinates than a longer, straighter stretch of road.